rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java

```diff
@@ -149,6 +149,8 @@ byte[] sendRequest(HttpSender.Request request) throws Throwable {
             });
         } catch (FailsafeException failsafeException) {
             throw failsafeException.getCause() == null ? failsafeException : failsafeException.getCause();
+        } catch (Throwable e) {
+            throw new HttpSenderResponseException(e, null, "");
```

Comment:
I worry a bit about how broadly we match any Throwable here, and then throw a `HttpSenderResponseException` despite the underlying issue having been an `UncheckedIOException`, with no guarantee a server response was involved here (as in your case the connection timed out). Elsewhere in `MavenPomDownloader` we use `HttpSenderResponseException` to signify that the server exists, with implications for caching that information beyond just the request here.

https://github.com/openrewrite/rewrite/blob/b67be0403ee9e5eea0e718642bda0ee7f0e8bbb6/rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java#L262-L267

Perhaps we should catch and unwrap the UncheckedIOException instead?

```suggestion
        } catch (UncheckedIOException e) {
            throw e.getCause();
```


That's the most likely case of unhandled exceptions looking at how we send requests

https://github.com/openrewrite/rewrite/blob/03c458404073082275a752063887c188d213afbe/rewrite-core/src/main/java/org/openrewrite/ipc/http/HttpUrlConnectionSender.java#L115-L117

Any thoughts on that?

Reply:
that does seem better -- will push some changes tomorrow

Reply:
Thanks! I've gone ahead and applied my suggestion, such that we can go into tomorrow knowing what effect this has on the tests already.